File: support_tools/swigwin-3.0.12/Examples/test-suite/go/using_composition_runme.go
package main

import . "./using_composition"

func main() {
	f := NewFooBar()
	if f.Blah(3).(int) != 3 {
		panic("FooBar::blah(int)")
	}

	if f.Blah(3.5) != 3.5 {
		panic("FooBar::blah(double)")
	}

	if f.Blah("hello").(string) != "hello" {
		panic("FooBar::blah(char *)")
	}

	f2 := NewFooBar2()
	if f2.Blah(3).(int) != 3 {
		panic("FooBar2::blah(int)")
	}

	if f2.Blah(3.5) != 3.5 {
		panic("FooBar2::blah(double)")
	}

	if f2.Blah("hello").(string) != "hello" {
		panic("FooBar2::blah(char *)")
	}

	f3 := NewFooBar3()
	if f3.Blah(3).(int) != 3 {
		panic("FooBar3::blah(int)")
	}

	if f3.Blah(3.5) != 3.5 {
		panic("FooBar3::blah(double)")
	}

	if f3.Blah("hello").(string) != "hello" {
		panic("FooBar3::blah(char *)")
	}
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/using_composition_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

using_composition

f = FooBar();
if (f.blah(3) != 3)
  error("FooBar::blah(int)")
endif

if (f.blah(3.5) != 3.5)
  error("FooBar::blah(double)")
endif

if (!strcmp(f.blah("hello"),"hello"))
  error("FooBar::blah(char *)")
endif


f = FooBar2();
if (f.blah(3) != 3)
  error("FooBar2::blah(int)")
endif

if (f.blah(3.5) != 3.5)
  error("FooBar2::blah(double)")
endif

if (!strcmp(f.blah("hello"),"hello"))
  error("FooBar2::blah(char *)")
endif


f = FooBar3();
if (f.blah(3) != 3)
  error("FooBar3::blah(int)")
endif

if (f.blah(3.5) != 3.5)
  error("FooBar3::blah(double)")
endif

if (!strcmp(f.blah("hello"),"hello"))
  error("FooBar3::blah(char *)")
endif

